### ISSUE TITLE
TorchVision

### ISSUE THREAD

**Issue by Ludobico** [Ludobico/Python_AL_01#25]
# torchvision.transforms
Pytorch의 torchvision 라이브러리는 transforms 에서 다양한 변환 기능을 제공합니다. transform을 사용하여 데이터의 일부 조작을 수행하고 훈련에 적합하게 만들 수 있습니다.

```py
from torchvision import transforms

data_transforms = {
    'train' : transforms.Compose([
        transforms.RandomResizedCrop(224),
        transforms.RandomHorizontalFlip(),
        transforms.ToTensor(),
        transforms.Normalize([0.485, 0.456, 0.406],
                             [0.229, 0.224, 0.225])
    ]),
    'val' : transforms.Compose([
        transforms.Resize(256),
        transforms.CenterCrop(224),
        transforms.ToTensor(),
        transforms.Normalize([0.485, 0.456, 0.406],
                             [0.229, 0.224, 0.225])
    ])
}
```

### Compose
transform에 여러 단계가 있을 경우, Compose를 통해 여러 단계를 하나로 묶을 수 있습니다. transforms에 속한 함수들은 Compose를 통해 묶어서 한번에 처리 할 수 있습니다.
```py
transforms.Compose([ 
   transforms.ToTensor(), 
   transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)) 
])
```

### ToTensor
ToTensor는 매우 일반적으로 사용되는 `conversion transform`입니다. PyTorch에서 우리는 주로 텐서 형태의 데이터로 작업합니다. 입력 데이터가 NumPy 배열 또는 PIL 이미지 형식인 경우 ToTensor를 사용하여 텐서 형식으로 변환할 수 있습니다.
```py
transforms.ToTensor()
```
### Resize
`transforms.resize`는 이미지의 크기를 조정하는 함수입니다. 이 함수는 기본적으로 이미지를 새로운 크기로 재조정할 때 사용합니다.
```py
transform = transforms.Resize(size, interpolation=2)
```

### Normalize
Normalize 작업은 텐서를 가져와 평균 및 표준 편차로 정규화합니다.
`transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)) - mean`
형식으로 평균을 입력하며, 괄호 안에 들어가 있는 수의 개수가 채널의 수입니다.

### CenterCrop
이것은 중앙에서 주어진 텐서 이미지를 자릅니다.
`transforms.CenterCrop( height, width )` 형식으로 자르려는 크기를 입력으로 제공할 수 있습니다.

### RandomHorizontalFlip
이 변환은 주어진 확률로 이미지를 무작위로 수평을 뒤집을(flip) 것입니다. 이 확률은 매개변수 `p`를 통해 설정할 수 있습니다. p의 기본값은 0.5입니다.

### RandomRotation
이 변환은 이미지를 각도만큼 무작위로 회전합니다. 도(degree) 단위의 각도는 해당 매개변수 `degree`에 대한 입력으로 제공될 수 있습니다.

### Grayscale
이 변환은 원본 RGB 이미지를 회색으로 변경합니다.
`num_output_channels` 매개변수에 입력으로 원하는 채널 수를 제공할 수 있습니다.

# torchvision.utils.make_grid()
torchvision.utils.make_grid()는 Pytorch의 공식 이미지 처리 라이브러리인 torchvision에서 제공하는 함수입니다. make_grid() 함수는 주어진 이미지들을 정해진 가로 세로 개수로 그리드 상에 배치하여 하나의 큰 이미지로 합칩니다. 가로 세로 개수는 입력 인자로 지정할 수 있습니다.

make_grid() 함수는 주어진 이미지들의 크기가 동일할 경우 주로 사용됩니다. 예를 들어, 이미지 분류기를 구현할 때 이미지 데이터셋의 샘플 이미지들을 그리드 상에 배치하여 전체 이미지를 시각화할 때 주로 사용합니다.

다음은 make_grid() 함수를 사용하는 예입니다.
```py
import torch
from torchvision import utils

# 이미지 배열을 생성합니다.
images = torch.zeros(4, 3, 256, 256)

# 이미지 배열을 그리드 상에 배치합니다.
grid = utils.make_grid(images, nrow=2)

# 생성된 이미지를 화면에 출력합니다.
utils.save_image(grid, 'grid.jpg')
```
